sdk/src/types/block/semantic/state_transition.rs

```diff
@@ -568,10 +705,18 @@ impl StateTransitionVerifier for DelegationOutput {
         next_state: &Self,
         context: &SemanticValidationContext<'_>,
     ) -> Result<(), TransactionFailureReason> {
-        Self::transition_inner(current_state, next_state)?;
+        if !current_state.delegation_id().is_null() || next_state.delegation_id().is_null() {
+            return Err(TransactionFailureReason::DelegationOutputTransitionedTwice);
+        }
 
-        let protocol_parameters = &context.protocol_parameters;
+        if current_state.delegated_amount() != next_state.delegated_amount()
+            || current_state.start_epoch() != next_state.start_epoch()
+            || current_state.validator_address() != next_state.validator_address()
+        {
+            return Err(TransactionFailureReason::DelegationModified);
+        }
 
+        let protocol_parameters = &context.protocol_parameters;
```

Comment:
Could just use `&context.protocol_parameters` in L724 since we only use it at this place, but feel free to resolve without doing if you think it's better for readability

Reply:
done